slasher/rpc/update_min_max_span.go

```diff
@@ -0,0 +1,75 @@
+package rpc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	ethpb "github.com/prysmaticlabs/prysm/proto/eth/v1alpha1"
+	"github.com/prysmaticlabs/prysm/shared/params"
+)
+
+// UpdateMaxSpan is used to update the max span of an incoming attestation after the slashing detection phase.
+// logic is following the detection method designed by https://github.com/protolambda
+// from here: https://github.com/protolambda/eth2-surround/blob/master/README.md#min-max-surround
+func (ss *Server) UpdateMaxSpan(ctx context.Context, source uint64, target uint64, validatorIdx uint64) error {
+	diff := target - source
+	if diff > params.BeaconConfig().WeakSubjectivityPeriod {
+		return fmt.Errorf("attestation detection supports only weak subjectivity period: %v target - source: %v > weakSubjectivityPeriod", params.BeaconConfig().WeakSubjectivityPeriod, diff)
+	}
+	spanMap, err := ss.SlasherDB.ValidatorSpansMap(validatorIdx)
+	if err != nil {
+		return errors.Wrapf(err, "could not retrieve span map for validatorIdx: %v", validatorIdx)
+	}
+	if spanMap.EpochSpanMap == nil {
```

Comment:
can you place this check in the slasher db ? This should be handled in the db

Reply:
thanks